pkg/api/devops/v1alpha3/pipeline_types.go

```diff
@@ -222,11 +224,24 @@ type DiscarderProperty struct {
 
 type ParameterDefinition struct {
 	Name         string `json:"name" description:"name of param"`
-	DefaultValue string `json:"default_value,omitempty" mapstructure:"default_value" description:"default value of param"`
+	DefaultValue string `json:"default_value,omitempty" yaml:"default_value" mapstructure:"default_value" description:"default value of param"`
+	IsQuoted     bool   `json:"isQuoted" yaml:"isQuoted" description:"whether this param is quoted to build dynamic"`
 	Type         string `json:"type" description:"type of param"`
 	Description  string `json:"description,omitempty" description:"description of pipeline"`
 }
 
+type ParameterReference struct {
+	corev1.TypedLocalObjectReference `json:",inline"`
+	ValuesKey                        string `json:"valuesKey" description:"reference resource data key"`
```

Comment:
Please following the example if you want make some fields be optional.

```
TimerTrigger      *TimerTrigger         `json:"timer_trigger,omitempty"
```

See the errors:
<img width="1323" alt="image" src="https://user-images.githubusercontent.com/1450685/163908528-294d1f7a-959a-4016-82f7-1a994713ff72.png">